internal/controllers/suite_test.go

```diff
@@ -162,6 +162,21 @@ func TestMain(m *testing.M) {
 		ErrorIfCRDPathMissing: true,
 	}
 
+	// ENVTEST-based tests require specific binaries. By default, these binaries are located
+	// in paths defined by controller-runtime. However, the `BinaryAssetsDirectory` needs
+	// to be explicitly set when running tests directly (e.g., debugging tests in an IDE)
+	// without using the Makefile targets.
+	//
+	// This is equivalent to configuring your IDE to export the `KUBEBUILDER_ASSETS` environment
+	// variable before each test execution. The following function simplifies this process
+	// by handling the configuration for you.
+	//
+	// To ensure the binaries are in the expected path without manual configuration, run:
+	// `make setup-envtest`
+	if getFirstFoundEnvTestBinaryDir() != "" {
```

Comment:
Hi @perdasilva,  

You added this comment, and I deviated—sorry about that!  

> Would it be cleaner if we just used an environment variable like `ENV_TEST_BINARY_DIR` or something like that?  
> Then, it doesn't really matter where the binaries are placed.  

We already have an environment variable (`KUBEBUILDER_ASSETS`) that can be used, but it’s not very convenient and involves a steep learning curve:  
- Know that ENVTEST requires binaries.
- Then, by default, ENVTEST is expected to find the bins in `local/...`.  
- That we can override this using the `KUBEBUILDER_ASSETS` environment variable.  
- Honestly, I’m not sure where Bingo was placing the specific bins used in the tests before.  
- Then, there’s the added challenge of knowing how to configure the tests properly in an IDE.  

This PR simplifies the process so everything “just works” without extra setup or deep knowledge. 
Making tests and debugging smoother and hassle-free.  

I’ve also updated the code comments to clarify things further.  
Hope this clears everything up! 😊

Reply:
Might also be nice to update the developer docs: https://github.com/operator-framework/operator-controller/blob/main/docs/contribute/developer.md

Reply:
Both are done 👍 
Thank you 🥇